components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -432,63 +442,300 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
         }
     }
 
-    private List<Object> filterUsers(Node node, Map<String, Boolean> requiredAttributes, int offset, int limit)
+    /**
+     * Filter users using multi-attribute filters or single attribute filters with pagination.
+     *
+     * @param node
+     * @param requiredAttributes
+     * @param offset
+     * @param limit
+     * @param sortBy
+     * @param sortOrder
+     * @param domainName
+     * @return
```

Comment:
lets have small description of each param and the return value

Reply:
Already fixed in later versions